src/core/CLucene/index/IndexWriter.cpp

```diff
@@ -1246,39 +1246,30 @@ void IndexWriter::indexCompaction(std::vector<lucene::store::Directory *> &src_d
     CND_CONDITION(dest_dirs.size() > 0, "Destination directory not found.");
     this->_trans_vec = std::move(trans_vec);
 
-    // order mapping: dir -> segment info -> segment reader
-    addIndexesSegments(src_dirs);
-
     // create segment readers
-    int32_t totDocCount = 0;
-    int numSegments = segmentInfos->size();
-    assert(numSegments > 0);
+    int numIndices = src_dirs.size();
 
     //Set of IndexReaders
     if (infoStream != NULL) {
-        message(string("src index dir size: ") + Misc::toString(numSegments));
+        message(string("src index dir size: ") + Misc::toString(numIndices));
     }
-    for (int32_t i = 0; i < numSegments; i++) {
-        SegmentInfo *si = segmentInfos->info(i);
-        IndexReader *reader = SegmentReader::get(si, MERGE_READ_BUFFER_SIZE, false /* mergeDocStores */);
+    for (int32_t i = 0; i < numIndices; i++) {
+        IndexReader* reader = lucene::index::IndexReader::open(src_dirs[i], MERGE_READ_BUFFER_SIZE, false);
```

Comment:
add comment to explain why change si to src_dir[i]

Reply:
Done.